src/wowman/core.clj

```diff
@@ -124,6 +129,66 @@
     (nav-map state path)
     (throw (RuntimeException. "application must be `start`ed before state may be accessed."))))
 
+(defn as-unqualified-hyphenated-maps
+  "used to coerce keys in each row of resultset
+  adapted from https://github.com/seancorfield/next-jdbc/blob/master/doc/result-set-builders.md#rowbuilder-and-resultsetbuilder"
+  [rs opts]
+  (let [xform #(-> % name lower-case (clojure.string/replace #"_" "-"))
+        unqualified (constantly nil)]
+    (rs/as-modified-maps rs (assoc opts :qualifier-fn unqualified :label-fn xform))))
+
+(defn db-query
+  ([query]
+   (db-query query nil))
+  ([query arg-list & {:keys [opts]}]
+   (jdbc/execute! (get-state :db) (into [query] arg-list)
+                  (merge {:builder-fn as-unqualified-hyphenated-maps} opts))))
+
+(def select-*-catalog (slurp "resources/query--all-catalog.sql"))
+
+(defn db-split-category-list
+  [category-list-str]
+  (if (empty? category-list-str)
+    []
+    (clojure.string/split category-list-str #"\|")))
+
+(defn db-gen-game-track-list
+  [row]
+  (let [track-list (vec (remove nil? [(when (:retail-track row) "retail")
+                                      (when (:vanilla-track row) "classic")]))
+        row (dissoc row :retail-track :vanilla-track)]
+    (if (empty? track-list)
+      row
+      (assoc row :game-track-list track-list))))
+
+(defn db-coerce-row-values
+  [row]
+  (when row
+    (->> row
+         (utils/coerce-row-values {:category-list db-split-category-list})
+         (db-gen-game-track-list))))
+
+(defn db-search
+  ([]
+   ;; random list of addons, no preference
+   (db-query (str select-*-catalog "order by RAND() limit ?") [(get-state :search-results-cap)]))
+  ([uin]
+   (let [uin% (str uin "%")
+         %uin% (str "%" uin "%")]
+     (sql/find-by-keys (get-state :db) :catalog ["label ilike ? or description ilike ?"
```

Comment:
this syntax is nice and friendly, but we're not getting the `:category-list` and `:game-track` wrangling

Reply:
indeed, if you do a search in the search pane, the categories disappear